build-tools/packages/build-cli/src/commands/vnext/exec.ts

```diff
@@ -0,0 +1,144 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IPackage } from "@fluid-tools/build-infrastructure";
+import { Args, Flags, ux } from "@oclif/core";
+import async from "async";
+import execa from "execa";
+
+import { releaseGroupNameFlag } from "../../flags.js";
+import { BaseCommandWithBuildProject } from "../../library/index.js";
+
+/**
+ * Runs a shell command in the context of packages within the BuildProject.
+ *
+ * This command is a vnext version that uses the build-infrastructure package APIs.
+ */
+export default class ExecCommand extends BaseCommandWithBuildProject<typeof ExecCommand> {
```

Comment:
The old command was a PackageCommand - maybe we need a BuildProjectPackageComand base for the vnext commands? It's nice to only have to implement the per-package handling.